### PR TITLE
Add deserializers for disaggregation protocol

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6827

Problem Summary:

### What is changed and how it works?

- [x] Add deserializers
- [x] Polished some names to be more understandable
- [ ] Use deserializers to rebuild the Segment snapshot

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/SharedContexts/Disagg.h [hunk 1]
```diff
@@ -45,11 +46,17 @@ struct SharedContextDisagg : private boost::noncopyable
 
     bool use_autoscaler = true; // TODO: remove this after AutoScaler is stable. Only meaningful in DisaggregatedComputeMode.
 
+    /// Only for read node.
```
Comment (JaySon-Huang) [inline dbms/src/Interpreters/SharedContexts/Disagg.h:49]: write node should also use remote_data_store to putDMFile?

**Review by JaySon-Huang — APPROVED** [review]

**Comment by flowbehappy** [comment]
LGTM

**Review by flowbehappy — APPROVED** [review]

**Pull request merged**